app/models/user.rb

```diff
@@ -40,6 +40,34 @@ def first_and_last
     "#{first_name} #{last_name_letter}."
   end
 
+  def win_streak
```

Comment:
Can we rename this to `win_loss_streak`?
The N+1 query is where your fetching the winner and loser records than grabbing the id. You can just to `winner_id`.
Here's a refactored method:
```
  def win_loss_streak
    return "-" if @complete_matches.blank?
    last_match_won = @complete_matches.last.winner_id == id
    streak = @complete_matches.reverse.take_while { |match| (id == match.winner_id && last_match_won) || (id != match.winner_id && !last_match_won) }.count
    "#{last_match_won ? "W" : "L"}#{streak}"
  end
```